libpod/pod.go

```diff
@@ -696,3 +710,35 @@ func (p *Pod) Status() (map[string]ContainerStatus, error) {
 // TODO add pod batching
 // Lock pod to avoid lock contention
 // Store and lock all containers (no RemoveContainer in batch guarantees cache will not become stale)
+
+// Inspect returns a PodInspect struct to describe the pod
+func (p *Pod) Inspect() (PodInspect, error) {
+	var (
+		podContainers []PodContainerInfo
+	)
+
+	containers, err := p.AllContainers()
+	if err != nil {
+		return PodInspect{}, err
+	}
+	for _, c := range containers {
+		containerStatus := "uknown"
+		// Ignoring possible errors here because we dont want this to be
+		// catastrophic in nature
+		containerState, err := c.State()
+		if err == nil {
+			containerStatus = containerState.String()
+		}
+		pc := PodContainerInfo{
+			ID:    c.ID(),
+			State: containerStatus,
+		}
+		podContainers = append(podContainers, pc)
+	}
+	inspectData := PodInspect{
+		Config:     p.config,
```

Comment:
This is really not safe - we're passing pointers to the actual libpod structs outside of libpod, so people can modify them. Better to deepcopier them and return copies

Reply:
deepcopier it is!